tests/test_form_max_fields_files_part_size.py

```diff
@@ -0,0 +1,89 @@
+from typing import List
+
+from fastapi import FastAPI, File, UploadFile
+from fastapi.testclient import TestClient
+
+app = FastAPI()
+
+
+@app.post("/", form_max_files=2, form_max_part_size=1024, form_max_fields=2)
+async def upload_files(files: List[UploadFile] = File(...)):
+    return {"filenames": [file.filename for file in files]}
+
+
+def test_form_max_files_send_one():
+    client = TestClient(app)
+
+    response = client.post(
+        "/",
+        files=[
+            ("files", ("file1.txt", b"file1 content", "text/plain")),
+        ],
+    )
+
+    assert response.status_code == 200, response.text
+    assert response.json() == {"filenames": ["file1.txt"]}
+
+
+def test_form_max_files_send_too_many():
+    client = TestClient(app)
+
+    response = client.post(
+        "/",
+        files=[
+            ("files", ("file1.txt", b"file1 content", "text/plain")),
+            ("files", ("file2.txt", b"file2 content", "text/plain")),
+            ("files", ("file3.txt", b"file3 content", "text/plain")),
+        ],
+    )
+
+    assert response.status_code == 400, response.text
+    assert response.json() == {
+        "detail": "Too many files. Maximum number of files is 2."
+    }
+
+
+def test_max_part_size_exceeds_custom_limit():
+    client = TestClient(app)
+
+    boundary = "------------------------4K1ON9fZkj9uCUmqLHRbbR"
+
+    multipart_data = (
+        f"--{boundary}\r\n"
+        f'Content-Disposition: form-data; name="small"\r\n\r\n'
+        "small content\r\n"
+        f"--{boundary}\r\n"
+        f'Content-Disposition: form-data; name="large"\r\n\r\n'
+        + ("x" * 1024 * 10 + "x")  # 1MB + 1 byte of data
```

Comment:
Would it make more sense to have it be `1KB +1` instead of `1MB +1`, to ensure the 1KB limit is precise, or am I misunderstanding something (quite possible!)?